### PR TITLE
[webthing] Fix item type for dynamic channel

### PR DESCRIPTION
**Pull request by lolodomo**
Related to #12712

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java [hunk 1]
```diff
@@ -31,12 +31,12 @@ public class TypeMapping {
...
     /**
      * maps a property type to an item type
-     * 
+     *
      * @param propertyMetadata the property meta data
      * @return the associated item type
      */
     public static ItemType toItemType(Property propertyMetadata) {
-        String type = "string";
+        String type = "String";
```
Comment (jlaur) [inline bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java:39]:
```suggestion
        String type = CoreItemFactory.STRING;
```

File: bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java [hunk 2]
```diff
@@ -48,70 +48,70 @@ public static ItemType toItemType(Property propertyMetadata) {
             case "MotionProperty":
             case "OnOffProperty":
             case "PushedProperty":
-                type = "switch";
+                type = "Switch";
```
Comment (jlaur) [inline bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java:51]:
```suggestion
                type = CoreItemFactory.SWITCH;
```

File: bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java [hunk 3]
```diff
@@ -48,70 +48,70 @@ public static ItemType toItemType(Property propertyMetadata) {
...
                 tag = "Switchable";
                 break;
             case "CurrentProperty":
             case "FrequencyProperty":
             case "InstantaneousPowerProperty":
             case "VoltageProperty":
-                type = "number";
+                type = "Number";
```
Comment (jlaur) [inline bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java:58]:
```suggestion
                type = CoreItemFactory.NUMBER;
```

Comment (jlaur) [reply bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java:58]: And so on... :-)

**Review by jlaur — CHANGES_REQUESTED** [review]
Looks good, just one minor suggested improvement opportunity to really ensure consistency.

File: bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java (hunk 3 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java:58]: Of course I can change here but as I checked all the bindings yesterday  I can tell you that most of them do not use these constants.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**